### PR TITLE
feat: Add example Python stack for Step Functions

### PR DESCRIPTION
**Pull request by lym953**
### What does this PR do?
Add an example Python stack that creates and instruments a State Machine.

### Testing Guidelines
Deployed the stack, and the traces showed up in Datadog app.
DatadogCdkPythonStack

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: examples/step-functions-python-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## Getting Started
+
+1. Get a Datadog API key to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
+2. Run `export DD_API_KEY=<DATADOG_API_KEY>` to set the Datadog API key in your shell session.
+3. Set up Datadog forwarder ([Datadog Forwarder documentation](https://docs.datadoghq.com/logs/guide/forwarder/?tab=cloudformation#installation)).
+4. Run `export DD_FORWARDER_ARN=<DD_FORWARDER_ARN>` to set the Datadog forwarder ARN in your shell session.
+5. Run `virtualenv env` to create a virtual environment.
+6. Run `source env/bin/activate` to activate the virtual environment.
+7. Run `pip install -r requirements.txt` install dependencies.
```
Comment (michaelcretzman) [inline examples/step-functions-python-stack/README.md:13]:
```suggestion
7. Run `pip install -r requirements.txt` to install dependencies.
```

**Review by michaelcretzman — CHANGES_REQUESTED** [review]

File: examples/step-functions-python-stack/README.md (hunk 1 above)
Comment (lym953) [reply examples/step-functions-python-stack/README.md]: Good catch!
Added `to` to this new doc and an existing doc.

**Review by avedmala — APPROVED** [review]
🚢

**Review by michaelcretzman — APPROVED** [review]

**Pull request merged**